src/modules/cpu/kernel/resample.hpp

```diff
@@ -0,0 +1,212 @@
+    window.pCenter = _mm_set1_ps(window.center);
+    window.pScale = _mm_set1_ps(window.scale);
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        Rpp32f *dstPtrTemp = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32f inRate = inRateTensor[batchCount];
+        Rpp32f outRate = outRateTensor[batchCount];
+        Rpp32s srcLength = srcLengthTensor[batchCount];
+        Rpp32s numChannels = channelTensor[batchCount];
+
+        if(outRate == inRate) {
+            // No need of Resampling, do a direct memcpy
+            memcpy(dstPtrTemp, srcPtrTemp, (size_t)(srcLength * numChannels * sizeof(Rpp32f)));
+        } else {
+            Rpp32s outBegin = 0;
+            Rpp32s outEnd = std::ceil(srcLength * outRate / inRate);
+            Rpp32s inPos = 0;
+            Rpp32s block = 1 << 8;
+            Rpp64f scale = (Rpp64f)inRate / outRate;
+            Rpp32f fscale = scale;
+
+            if(numChannels == 1) {
+                for (int outBlock = outBegin; outBlock < outEnd; outBlock += block) {
+                    Rpp32s blockEnd = std::min(outBlock + block, outEnd);
+                    Rpp64f inBlockRaw = outBlock * scale;
+                    Rpp32s inBlockRounded = std::floor(inBlockRaw);
+                    Rpp32f inPos = inBlockRaw - inBlockRounded;
+                    const Rpp32f * __restrict__ inBlockPtr = srcPtrTemp + inBlockRounded;
+
+                    for (int outPos = outBlock; outPos < blockEnd; outPos++, inPos += fscale) {
+                        Rpp32s loc0, loc1;
+                        window.input_range(inPos, &loc0, &loc1);
+                        if (loc0 + inBlockRounded < 0)
+                            loc0 = -inBlockRounded;
+                        if (loc1 + inBlockRounded > srcLength)
+                            loc1 = srcLength - inBlockRounded;
+                        Rpp32f accum = 0.0f;
+                        Rpp32s locInWindow = loc0;
+
+                        __m128 pAccum = xmm_p0;
+                        __m128 pLocInWindow = _mm_setr_ps(locInWindow - inPos, locInWindow + 1 - inPos, locInWindow + 2 - inPos, locInWindow + 3 - inPos);
+                        for (; locInWindow + 3 < loc1; locInWindow += 4) {
+                            __m128 w4 = window(pLocInWindow);
+
+                            pAccum = _mm_add_ps(pAccum, _mm_mul_ps(_mm_loadu_ps(inBlockPtr + locInWindow), w4));
+                            pLocInWindow = _mm_add_ps(pLocInWindow, xmm_p4);
+                        }
+
+                        pAccum = _mm_add_ps(pAccum, _mm_shuffle_ps(pAccum, pAccum, _MM_SHUFFLE(1, 0, 3, 2)));
+                        pAccum = _mm_add_ps(pAccum, _mm_shuffle_ps(pAccum, pAccum, _MM_SHUFFLE(0, 1, 0, 1)));
+                        accum = _mm_cvtss_f32(pAccum);
```

Comment:
Please check online if there is any better way to do addition of all sss vector values instead of this method followed from 157 - 159

Reply:
Couldn't find any better alternative